consensus/round/round.go

```diff
@@ -84,27 +78,57 @@ func NewRound(args ArgsRound) (*round, error) {
 
 // UpdateRound updates the index and the time stamp of the round depending on the genesis time and the current time given
 func (rnd *round) UpdateRound(genesisTimeStamp time.Time, currentTimeStamp time.Time) {
-	if !common.IsSupernovaRoundActivated(rnd.enableEpochsHandler, rnd.enableRoundsHandler) {
-		rnd.updateRound(genesisTimeStamp, currentTimeStamp, rnd.startRound)
-		return
+	baseTimeStamp := rnd.supernovaGenesisTimeStamp
+	roundDuration := rnd.supernovaTimeDuration
+	startRound := rnd.supernovaStartRound
+
+	supernovaActivated := rnd.isSupernovaActivated(currentTimeStamp)
+
+	if !supernovaActivated {
+		baseTimeStamp = genesisTimeStamp
+		roundDuration = rnd.timeDuration
+		startRound = rnd.startRound
+	}
+
+	rnd.updateRound(baseTimeStamp, currentTimeStamp, startRound, roundDuration)
+}
+
+func (rnd *round) isSupernovaRoundActivated() bool {
+	return rnd.enableRoundsHandler.IsFlagEnabledInRound(common.SupernovaRoundFlag, uint64(rnd.index))
+}
+
+func (rnd *round) isSupernovaActivated(currentTimeStamp time.Time) bool {
+	supernovaActivated := rnd.isSupernovaRoundActivated()
+	if supernovaActivated {
+		return supernovaActivated
+	}
+
+	currentTimeAfterSupernova := currentTimeStamp.UnixMilli() > rnd.supernovaGenesisTimeStamp.UnixMilli()
```

Comment:
can you check if this should be instead `>=` ?
maybe create a unit test for that